test cases/failing/123 subproject sandbox violation/meson.build

```diff
@@ -0,0 +1,34 @@
+project('subproject-sandbox-violation')
+
+sub1_d = subproject('subproj1').get_variable('d')
+sub1_mustfail = sub1_d.get_variable('dir') / '..' / 'file.txt'
+
+sub2_d = subproject('subproj2').get_variable('d')
+sub2_mustfail = sub2_d.get_variable('dir') / 'file.txt'
+
+main_d = declare_dependency(
+    variables: [
+        'dir=@0@'.format(meson.current_source_dir()),
+    ]
+)
+main_mustfail = main_d.get_variable('dir') / 'subprojects/subproj3/file.txt'
+
+if get_option('failmode') == 'parent-dir'
+    mustfail = sub1_mustfail
+elif get_option('failmode') == 'not-installed'
+    mustfail = sub2_mustfail
+elif get_option('failmode') == 'root-subdir'
+    mustfail = main_mustfail
+endif
+
+custom_target(
+    'mustfail',
+    input: mustfail,
+    output: 'file.txt',
+    command: [
+        'python3', '-c',
```

Comment:
This should use the Python module because `python3` is not available in plain `cmd.exe` (which we sadly don't have in our CI setup). See #10305 for a similar change for the `echo` command.

Reply:
`custom_target(..., command: ['string'])` internally runs `self.find_program_impl(['string'])`, and find_program_impl already does fallback for `python3` -> `mesonlib.python_command`.

I prefer to not use the python module as that costs a module import, then does obscure things to set up extension_module introspection data.